src/server-app/editor-server.html

```diff
@@ -0,0 +1,159 @@
+<!--
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<link rel="import" href="../../bower_components/polymer/polymer-element.html">
+
+<dom-module id="editor-server">
+  <template>
+    <style>
+          :host {
+            display: block;
+            height: 100%;
+          }
+
+          #container {
+            height: 100%;
+          }
+
+           iframe {
+            width: 100%;
+            height: 100%;
+            border: none;
+            top: 0;
+            opacity: 1;
+            transition: -webkit-filter 500ms, opacity 300ms;
+            overflow: hidden;
+          } 
+
+          iframe.gray {
+            -webkit-filter: grayscale(100%);
+            z-index: 1;
+          }
+
+          iframe.fade {
+            opacity: 0;
+          }
+
+    </style>
+    
+    <div id="container"> 
+    </div>
+
+    
+  </template>
+
+  <script src="../../node_modules/socket.io-client/dist/socket.io.js"></script>
+
+  <script>
+    /**
+     * @customElement
+     * @polymer
+     */
+    class EditorServer extends Polymer.Element {
+      static get is() { return 'editor-server'; }
+
+      constructor() {
+        super();
+        window.addEventListener('message', this._onMessage.bind(this));
+      }
+
+      ready() {
+        super.ready();
+        if (!window.parent) return;
+
+        this.socket = io();     
+        this.socket.emit('client-connection');
```

Comment:
This is the part where the server should check the cookie and associate cookie with socket id